Bottom: d8966719c9154c255d55c627507e07ee844d397a
Top:    bb373dec03a23765c5924386b8b621fc2f36679e
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 15:00:57 +0800

autofs-5.0.7 - fix incorrect check in flag_is_owned()

The flag file code isn't used any more but this is clearly incorrect
so fix it in case it gets used sometime in the future.


---

diff --git a/daemon/flag.c b/daemon/flag.c
index f8fe163..db9a4bd 100644
--- a/daemon/flag.c
+++ b/daemon/flag.c
@@ -66,12 +66,11 @@ static int flag_is_owned(int fd)
 
 			continue;
 		}
-
-		/* Stale flagfile */
-		if (!tries)
-			return 0;
 	}
 
+	/* Stale flagfile */
+	if (!tries)
+		return 0;
 
 	if (pid) {
 		int ret;
